src/components/elements/table/TableRowActions.tsx

```diff
@@ -0,0 +1,63 @@
+import { Stack } from '@mui/material';
+import { useMemo } from 'react';
+import ButtonLink from '../ButtonLink';
+import CommonMenuButton, { CommonMenuItem } from '../CommonMenuButton';
+
+export type TableRowAction<T> = Omit<CommonMenuItem, 'to' | 'onClick'> & {
+  // getUrl: given a record, return the URL this action should link to.
+  // E.g. for the "View Client" action on an Enrollment table, given the Enrollment record, return the link to the Client.
+  // getUrl could be made optional and onClick (omitted above) could be re-enabled
+  getUrl: (record: T) => string;
```

Comment:
API changes to this component? 
- Sometimes (like in bulk services) it needs to depend more on the row, be more configurable. 
  - Should be able to pass an action button directly like a React node?
  - getTableRowActions, similar to getColumnDefs ?
- Passing the primaryAction and secondaryActions so that you can have secondary only
- preferable to extend our current api to keep using rowLinkTo? I think this is a bit misleading (and also ambiguous what the record linking to the row should be named)

```
interface GenericTableProps<T> {
  ...
  getTableRowActions?: (row: T) => TableRowAction<T>[] | ReactNode;
}
```

or maybe just get rid of `TableRowAction<T>[]` and have it always return `ReactNode`?